lib/beetle/message.rb

```diff
@@ -362,5 +364,18 @@ def ack!
         @store.del_keys(msg_id)
       end
     end
+
+    def exponential_back_off?
+      @exp_bo
+    end
+
+    def calculate_next_delay(n, base_delay)
+      unless exponential_back_off?
+        base_delay
+      else
+        base_delay_n = base_delay * (2**n)
+        rand((base_delay_n - base_delay_n / 4.0)..(base_delay_n + base_delay_n / 2.0))
```

Comment:
I'm not sure that picking a random number within the backoff interval has any effect and I'd rather keep the code simple. What's missing is an upper bound for the delay. A max_delay option. See ZMQ_RECONNECT_IVL and ZMQ_RECONNECT_IVL_MAX for zeromq http://api.zeromq.org/2-1:zmq-setsockopt.